server/app/com/xsn/explorer/databaseMigrations/BlockRewardsMigration.scala

```diff
@@ -0,0 +1,126 @@
+package com.xsn.explorer.databaseMigrations
+
+import com.alexitc.playsonify.core.FutureOr.Implicits.FutureOps
+import com.xsn.explorer.data.anorm.AnormPostgresDataHandler
+import com.xsn.explorer.data.anorm.dao.{BlockPostgresDAO, BlockRewardPostgresDAO}
+import com.xsn.explorer.data.async.BlockFutureDataHandler
+import com.xsn.explorer.executors.DatabaseExecutionContext
+import com.xsn.explorer.models.{BlockExtractionMethod, BlockRewards, rpc}
+import com.xsn.explorer.models.persisted.Block
+import com.xsn.explorer.models.values.{Blockhash, Height}
+import com.xsn.explorer.services.{BlockService, XSNService}
+import javax.inject.Inject
+import org.scalactic.{Bad, Good}
+import org.slf4j.LoggerFactory
+import play.api.db.Database
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class BlockRewardsMigration @Inject()(
+    xsnService: XSNService,
+    blockService: BlockService,
+    blockDataHandler: BlockFutureDataHandler,
+    db: BlockRewardsMigration.DatabaseOperations
+)(implicit ec: ExecutionContext) {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  def run() = {
+    val targetBlock = blockDataHandler.getLatestBlock().toFutureOr
+
+    targetBlock.map { targetBlock =>
+      logger.info(s"Migrating Block Rewards from block 0 to block ${targetBlock.height}")
+
+      val startingState = Future.successful(Good(())).toFutureOr
+      val finalState = (0 to targetBlock.height.int).foldLeft(startingState) {
+        case (state, height) =>
+          for {
+            _ <- state
+            block <- blockDataHandler.getBy(Height(height)).toFutureOr
+            rpcBlock <- xsnService.getFullBlock(block.hash).toFutureOr
+            _ <- block.extractionMethod match {
+              case BlockExtractionMethod.TrustlessProofOfStake =>
+                recalculateExtractionMethod(block, rpcBlock).toFutureOr
+              case _ => Future.successful(Good(())).toFutureOr
+            }
+            _ <- db.getReward(block.hash).toFutureOr.map {
```

Comment:
This is not working as you likely expect, you'll end up with `FutureOr[Future[_]]`, use `flatMap` instead of `map`